content/warnings/invalid-hook-call-warning.md

```diff
@@ -0,0 +1,121 @@
+
+## Breaking the Rules of Hooks
+
+You can only call Hooks **while React is rendering a function component**:
+
+* ✅ Call them at the top level in the body of a function component.
+* ✅ Call them at the top level in the body of a [custom Hook](/docs/hooks-custom.html).
+
+**Learn more about this in the [Rules of Hooks](/docs/hooks-rules.html).**
+
+To avoid confusion, it’s **not** supported to call Hooks in other cases:
+
+* 🔴 Do not call Hooks in class components.
+* 🔴 Do not call in event handlers.
+* 🔴 Do not call Hooks inside functions passed to `useMemo`, `useReducer`, or `useEffect`.
+
+If you break these rules, you might see this error.
+
+```js{2-3,8-9,15-16,23-24,33-34}
+function Counter() {
+  // ✅ Good: top-level in a function component
+  const [count, setCount] = useState(0);
+  // ...
+}
+
+function useWindowWidth() {
+  // ✅ Good: top-level in a custom Hook
+  const [width, setWidth] = useState(window.innerWidth);
+  // ...
+}
+
+function Bad1() {
+  function handleClick() {
+    // 🔴 Bad: inside an event handler (to fix, move it outside!)
+    const theme = useContext(ThemeContext);
+  }
+  // ...
+}
+
+function Bad2() {
+  const style = useMemo(() => {
+    // 🔴 Bad: inside useMemo (to fix, move it outside!)
+    const theme = useContext(ThemeContext);
+    return createStyle(theme);
+  });
+  // ...
+}
+
+
+class Bad3 extends React.Component {
+  render() {
+    // 🔴 Bad: inside a class component
+    useEffect(() => {})
+    // ...
+  }
+}
+```
+
+You can use the [`eslint-plugin-react-hooks` plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) to catch some of these mistakes.
+
+>Note
+>
+>[Custom Hooks](/docs/hooks-custom.html) *may* call other Hooks (that's their whole purpose). This works because custom Hooks are also supposed to only be called while a function component is rendering.
+
+
+## Duplicate React
+
+In order for Hooks to work, `react` import from your application code needs to resolve to the same module as the `react` import from inside the `react-dom` package.
+
+If these `react` imports resolve to two different exports objects, you will see this warning. This may happen if you **accidentally end up with two copies** of the `react` package.
+
+If you use Node for package management, you can run this check in your project folder:
+
+    npm ls react
+
+If you see more than one React, you'll need to figure out why this happens, and fix your dependency tree. For example, maybe a library you're using incorrectly specifies `react` as a dependency (rather than a peer dependency). Until such a library is fixed, [Yarn resolutions](https://yarnpkg.com/lang/en/docs/selective-version-resolutions/) is one possible workaround.
```

Comment:
```suggestion
If you see more than one React, you'll need to figure out why this happens, and fix your dependency tree. For example, maybe a library you're using incorrectly specifies `react` as a dependency (rather than a peer dependency). Until that library is fixed, [Yarn resolutions](https://yarnpkg.com/lang/en/docs/selective-version-resolutions/) is one possible workaround.
```